.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,108 @@
+name: Build and Unit Test Checks
+on:
+  pull_request:
+    branches: [ master, develop ]
+    types: ['opened', 'reopened', 'synchronize']
+    paths-ignore:
+    - 'README.md'
+
+jobs:
+  build_sdk:
+    name: Build Check for the .NET SDK
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 5.0.x
+    - name: Install dependencies
+      run: dotnet restore RudderAnalytics/RudderAnalytics.csproj
+    - name: Build
+      run: dotnet build RudderAnalytics/RudderAnalytics.csproj --configuration Release --no-restore
+  build_and_test_netcoreapp21:
+    name: Build and Quality Check for .NetCoreApp2.1
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 2.1.x
+    - name: Install dependencies
+      run: dotnet restore Test/Test.csproj
+    - name: Build
+      run: dotnet build Test/Test.csproj --configuration Release --no-restore
+    - name: Test      
+      run: dotnet test Test/Test.csproj --no-restore --verbosity normal
+  build_and_test_netstandard20:
+    name: Build and Quality Check for .NetStandard20
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 2.1.x
+    - name: Install dependencies
+      run: dotnet restore Test.NetStandard20/Test.NetStandard20.csproj
+    - name: Build
+      run: dotnet build Test.NetStandard20/Test.NetStandard20.csproj --configuration Release --no-restore
+    - name: Test      
+      run: dotnet test Test.NetStandard20/Test.NetStandard20.csproj --no-restore --verbosity normal
+  build_and_test_net50:
+    name: Build and Quality Check for .Net50
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 5.0.x
+    - name: Install dependencies
+      run: dotnet restore Test.Net50/Test.Net50.csproj
+    - name: Build
+      run: dotnet build Test.Net50/Test.Net50.csproj --configuration Release --no-restore
+    - name: Test      
+      run: dotnet test Test.Net50/Test.Net50.csproj --no-restore --verbosity normal
```

Comment:
dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura to generate coverage report 

Abhishek also has a step in order to upload the reports in the build results

Reply:
done